solace-message-router/azuredeploy.json

```diff
@@ -5,57 +5,117 @@
       "metadata": {
-        "description": "Security group defined to support VMR default message vpn ports"
+        "description": "Security group defined to support VMR system level and default message vpn ports."
       }
     },
-    "dnsLabelForPublicIP": {
+    "workspaceName": {
       "type": "string",
+      "defaultValue": "",
       "metadata": {
-        "description": "Unique DNS Label for the Public IP used to access the Virtual Machine."
+        "description": "Unique OMS Workspace Name for Log Analytics. Leave this field empty to not deploy an OMS Workspace."
       }
     },
-    "centOSVersion": {
+    "workspaceRegion": {
       "type": "string",
-      "defaultValue": "7.2",
+      "defaultValue": "East US",
       "metadata": {
-        "description": "The Centos version for deploying the Docker containers. This will pick a fully patched image of this given Centos version. Allowed values: 7.2, 7.3"
+        "description": "OMS Workspace Region for Log Analytics. Not used if Workspace Name is empty."
+      },
+      "allowedValues": [
+        "East US",
+        "West Europe",
+        "Southeast Asia",
+        "Australia Southeast",
+        "Japan East",
+        "UK South",
+        "Central India",
+        "Canada Central"
+      ]
+    },
+    "dnsLabelForLbIp": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Label for the Public IP used to access the Load Balancer."
+      }
+    },
+    "dnsLabelForVmIp": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Label for the Public IP used to access the Virtual Machine(s)."
+      }
+    },
+    "centosVersion": {
+      "type": "string",
+      "defaultValue": "7.4",
+      "metadata": {
+        "description": "The CentOS version for deploying the Docker containers. This will pick a fully patched image of this given CentOS version. Allowed values: 7.2, 7.3, 7.4"
       },
       "allowedValues": [
         "7.2",
-        "7.3"
+        "7.3",
+        "7.4"
       ]
     },
-    "vmSize": {
+    "messageRoutingVmSize": {
       "type": "string",
       "defaultValue": "Standard_D2_v2",
       "metadata": {
-        "description": "The size of the VM. Requires at lest 2 cores and 4GB of memory. Allowed values: DS2_V2, F2S"
+        "description": "The size of the Solace Message Routing Node VM. Requires at least 2 cores and 4GB of memory. Allowed values: D2_v2, DS2_v2, D2_v3, D2s_v3"
       },
       "allowedValues": [
         "Standard_D2_v2",
-        "Standard_F2s"
+        "Standard_DS2_v2",
+        "Standard_D2_v3",
+        "Standard_D2s_v3"
       ]
     },
-    "solaceVMR_URI": {
+    "monitorVmSize": {
+      "type": "string",
+      "defaultValue": "Standard_D2_v2",
+      "metadata": {
+        "description": "The size of the Solace Monitor Node VM. Requires at least 2 cores and 4GB of memory. Allowed values: D2_v2, DS2_v2, D2_v3, D2s_v3"
+      },
+      "allowedValues": [
```

Comment:
remove the allowedValues altogether

Reply:
We have a minimum core and memory requirement for our docker container, so not every VM size will work. This is the subset of VM sizes we have tested and work, and we are willing to maintain this list over time.